topgun/k8s/kubernetes_creds_mgmt_test.go

```diff
@@ -88,48 +87,106 @@ var _ = Describe("Kubernetes credential management", func() {
 
 			fly.Run("unpause-pipeline", "-p", "pipeline")
 
-		})
-
-		var runsBuildWithCredentialsResolved = func() {
 			session := fly.Start("trigger-job", "-j", "pipeline/unit", "-w")
 			Wait(session)
 
+			By("seeing the credentials were resolved by concourse")
 			Expect(string(session.Out.Contents())).To(ContainSubstring("bar"))
 			Expect(string(session.Out.Contents())).To(ContainSubstring("zaz"))
 		}
 
-		Context("using the default namespace created by the chart", func() {
-			It("succeeds", runsBuildWithCredentialsResolved)
-		})
+		var runGetsCachedCredentials = func(normalSecret string, specialKeySecret string) {
+			runsBuildWithCredentialsResolved(normalSecret, specialKeySecret)
+			deleteSecret(releaseName, "main", normalSecret)
+			deleteSecret(releaseName, "main", specialKeySecret)
+			By("seeing that concourse uses the cached credentials")
+			runsBuildWithCredentialsResolved(normalSecret, specialKeySecret)
+		}
+
+		var cachingSetup = func() {
+			extraArgs = []string{
+				"--set=concourse.web.secretCacheEnabled=true",
+				"--set=concourse.web.secretCacheDuration=600",
+			}
+		}
+
+		var disableTeamNamspaces = func() {
+			By("creating a namespace made by the user instead of the chart")
+			Run(nil, "kubectl", "create", "namespace", releaseName+"-main")
+			extraArgs = []string{
+				"--set=concourse.web.kubernetes.createTeamNamespaces=false",
+			}
+		}
+
+		var resetExtraArgs = func() {
+			extraArgs = []string{}
+		}
+
+
+		Context("using per-team credentials", func() {
+			normalSecret := "foo"
+			specialKeySecret := "caz"
```

Comment:
Hey, wdyt of being more explicit in terms of naming here? E.g., what's "special" about it that "normal" doesn't have? 

Thanks!

Reply:
Naming hard. The only thing different about `specialKeySecret` is that instead of it's key in the kv-map isn't `value` it's `baz`.
We'll think on it a bit, we weren't too happy with it either.

Reply:
Renamed these. wdyt?